tests/auto-alex-v3.test.ts

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+import { Cl, ClarityType } from '@stacks/transactions';
+import { beforeEach, describe, expect, it } from 'vitest';
+import {
+    createClientMockSetup
+} from './clients/mock-client';
+import { tx } from '@hirosystems/clarinet-sdk';
+
+const { contracts, oracle, user2, bot, operator3, prepareALEX } = createClientMockSetup();
+
+describe('auto-alex-v3', () => {
+    it('auto-alex-v3 : ensure that it works', () => {
+
+        const wallet_1 = oracle;
+        const wallet_2 = bot;
+        const wallet_3 = user2;
+        const wallet_4 = operator3;
+        const dx = 1e8;
+        const end_cycle = 32;
+        const ACTIVATION_BLOCK = 20;
+
+        let response = prepareALEX();
+        expect(response.result).toHaveClarityType(ClarityType.ResponseOk);
+
+        if (simnet.blockHeight < ACTIVATION_BLOCK) {
+            simnet.mineEmptyBlocks(ACTIVATION_BLOCK - simnet.blockHeight);
+        }
+
+        // add some legacy auto-alex-v2 position and new auto-alex-v3 positions.
+        response = simnet.mineBlock([
+            tx.callPublicFn(contracts.autoAlexV2, "add-to-position", [Cl.uint(dx)], wallet_1),
+            tx.callPublicFn("auto-alex-v3-endpoint", "add-to-position", [Cl.uint(dx)], wallet_2),
+            tx.callPublicFn("auto-alex-v3-endpoint", "add-to-position", [Cl.uint(dx)], wallet_3),
+            tx.callPublicFn("auto-alex-v3-endpoint", "add-to-position", [Cl.uint(dx)], wallet_4)
+        ]);
+        response.map((e: any) => expect(e.result).toHaveClarityType(ClarityType.ResponseOk));
+
+        // move a few (2) cycles
+        simnet.mineEmptyBlocks(ACTIVATION_BLOCK + 2 * 525 - simnet.blockHeight);
+        
+        response = simnet.mineBlock([
+            // this fixes a minor bug in auto-alex-v2 that bounty must be greater than 0
+            tx.callPublicFn(contracts.oldAlex, 'mint-fixed', [Cl.uint(3), Cl.principal(contracts.autoAlexV2)], simnet.deployer),
+            // rebase the lialex
+            tx.callPublicFn('auto-alex-v3-endpoint', 'rebase', [], simnet.deployer),
```

Comment:
What is the expected effect of this tx?

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/20107362b60f7890c0538305e7089746f26b6ac3